src/logger.sh

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+# Copyright (c) YugaByte, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.  See the License for the specific language governing permissions and limitations
+# under the License.
+#
+
+[[ "${_YB_LOGGER_INCLUDED:=""}" == "yes" ]] && return 0
+_YB_LOGGER_INCLUDED=yes
+
+# -------------------------------------------------------------------------------------------------
+# Global variables used in this module
+# -------------------------------------------------------------------------------------------------
+yb_fatal_quiet=${yb_fatal_quiet:-false}
+yb_log_quiet=${yb_log_quiet:-false}
+FAIL_ON_WARNING=${FAIL_ON_WARNING:-false}
+
+# -------------------------------------------------------------------------------------------------
+# Global variables defined in this module
+# -------------------------------------------------------------------------------------------------
+# shellcheck disable=SC2034,SC2155
+readonly YB_BASH_COMMON_ROOT=$( cd "${BASH_SOURCE/*}" && cd .. && pwd )
+
+# shellcheck disable=SC2034,SC2155
+readonly YELLOW_COLOR="\033[0;33m"
+
+# shellcheck disable=SC2034,SC2155
+readonly RED_COLOR="\033[0;31m"
+
+# shellcheck disable=SC2034,SC2155
+readonly CYAN_COLOR="\033[0;36m"
+
+# shellcheck disable=SC2034,SC2155
+readonly NO_COLOR="\033[0m"
+
+# shellcheck disable=SC2034,SC2155
+# http://stackoverflow.com/questions/5349718/how-can-i-repeat-a-character-in-bash
+readonly HORIZONTAL_LINE=$( printf '=%.0s' {1..80} )
```

Comment:
Interestingly, this will cause an external process call, at least in RHEL 8:
```
# which printf
/bin/printf
```
And we are not really using this, so maybe we should get rid of this variable (or hard-code a string of 80 characters without calling external programs).